src/pages/home/report/ReportActionItemMessageEdit.js

```diff
@@ -0,0 +1,129 @@
+    ...windowDimensionsPropTypes,
+};
+
+class ReportActionItemMessageEdit extends React.Component {
+    constructor(props) {
+        super(props);
+        this.updateDraft = this.updateDraft.bind(this);
+        this.deleteDraft = this.deleteDraft.bind(this);
+        this.debouncedSaveDraft = _.debounce(this.debouncedSaveDraft.bind(this), 1000, true);
+        this.publishDraft = this.publishDraft.bind(this);
+        this.triggerSaveOrCancel = this.triggerSaveOrCancel.bind(this);
+
+        this.state = {
+            draft: this.props.draftMessage,
+        };
+    }
+
+    /**
+     * Update the value of the draft in Onyx
+     *
+     * @param {String} newDraft
+     */
+    updateDraft(newDraft) {
+        const trimmedNewDraft = newDraft.trim();
+        this.setState({draft: trimmedNewDraft});
+        this.debouncedSaveDraft(trimmedNewDraft);
+    }
+
+    /**
+     * Delete the draft of the comment being edited. This will take the comment out of "edit mode" with the old content.
+     */
+    deleteDraft() {
+        saveReportActionDraft(this.props.reportID, this.props.action.reportActionID, '');
+        toggleReportActionComposeView(true, this.props.isSmallScreenWidth);
+    }
+
+    /**
+     * Save the draft of the comment. This debounced so that we're not ceaselessly saving your edit. Saving the draft
+     * allows one to navigate somewhere else and come back to the comment and still have it in edit mode.
+     */
+    debouncedSaveDraft() {
+        saveReportActionDraft(this.props.reportID, this.props.action.reportActionID, this.state.draft);
+    }
+
+    /**
+     * Save the draft of the comment to be the new comment message. This will take the comment out of "edit mode" with
+     * the new content.
+     */
+    publishDraft() {
+        editReportComment(this.props.reportID, this.props.action, this.state.draft);
+        this.deleteDraft();
+    }
+
+    /**
+     * Key event handlers that short cut to saving/canceling.
+     *
+     * @param {Event} e
+     */
+    triggerSaveOrCancel(e) {
+        if (e && e.key === 'Enter' && !e.shiftKey) {
+            e.preventDefault();
+            this.publishDraft();
+        } else if (e && e.key === 'Escape') {
+            e.preventDefault();
+            this.deleteDraft();
+        }
+    }
+
+    render() {
+        return (
+            <View style={styles.chatItemMessage}>
+                <TextInputFocusable
+                    multiline
+                    onChangeText={this.updateDraft} // Debounced saveDraftComment
+                    onKeyPress={this.triggerSaveOrCancel}
+                    defaultValue={this.props.draftMessage}
+                    maxLines={16} // This is the same that slack has
+                    style={[styles.textInput, styles.flex0]}
+                    onFocus={() => {
+                        scrollToIndex({animated: true, index: this.props.index}, true);
+                        toggleReportActionComposeView(false);
```

Comment:
Since scrolling and toggling of the composer could result in a race condition due to animations, it is better to wait for the animations to complete before scrolling. This caused an issue #40767 where the composer box is partially hidden.